src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -325,10 +321,6 @@ public static KDF getInstance(String algorithm, Provider provider)
      * @throws NoSuchAlgorithmException
      *         if no {@code Provider} supports a {@code KDF} implementation for
      *         the specified algorithm
-     * @throws InvalidAlgorithmParameterException
-     *         if at least one {@code Provider} supports a {@code KDF}
-     *         implementation for the specified algorithm but none of them
-     *         support the specified parameters
```

Comment:
I don't understand why this @throws has been removed. @wangweij did you request that this be removed? What about the case where there is a provider that supports the algorithm, but does not support the parameters? I thought we were going to iterate over the providers and try to instantiate each one with the supplied parameters. In that case `NoSuchAlgorithmException` is not appropriate.

Reply:
Not me.

Reply:
This is the by-product of the `getNext()` refactoring. I should have modeled `getNext()` using the code under `getInstance()` instead of `chooseProvider()` so the debugging and exception can be preserved. Shared the prototype with Kevin in a haste and just mentioned that additional code may be needed for debugging.

Reply:
IAPE should be preserved. `getNext()` needs to be updated to preserve and throw IAPE to address this.